src/Native/Unix/System.Native/pal_networking.cpp

```diff
@@ -915,13 +919,37 @@ extern "C" int32_t SystemNative_GetDomainName(uint8_t* name, int32_t nameLength)
     assert(name != nullptr);
     assert(nameLength > 0);
 
+#if HAVE_GETDOMAINNAME
 #if HAVE_GETDOMAINNAME_SIZET
     size_t namelen = UnsignedCast(nameLength);
 #else
     int namelen = nameLength;
 #endif
 
     return getdomainname(reinterpret_cast<char*>(name), namelen);
+#elif HAVE_UNAME
+    // On Android, there's no getdomainname but we can use uname to fetch the domain name
+    // of the current device
+    size_t namelen = UnsignedCast(nameLength);
+    utsname  uts;
+
+    // If uname returns an error, bail out.
+    if(uname(&uts) == -1) {
```

Comment:
Nit: for here and below, the style we generally try to file in this file and others is to have a space between the if and the opening paren, and to have the opening brace on the following line